fggs/fggs.py

```diff
@@ -573,18 +597,17 @@ def __init__(self, start: Union[EdgeLabel, str, None]):
     def from_hrg(hrg: HRG):
         """Create an FGG out of an HRG and no domains and factors."""
         fgg = FGG(hrg.start)
-        for r in hrg.all_rules():
-            fgg.add_rule(r)
+        fgg._node_labels = hrg._node_labels.copy()
```

Comment:
So does all of this mean that a FactorGraph or FGG won't share a labeling with its Graph or HRG?

Reply:
Yes, I followed the same behavior as `copy()`.

Generally I think it's good to be conservative about sharing labelings. An HRG shares its labeling with its rules, and an FGG shares its labeling with its rules, and I think it's good if that's it. More complicated sharing setups could lead to inadvertently letting edge_labels become bigger than expected, which could cause slowdowns and unexpected conflicts (both of which are potential objections to this PR in general).

Reply:
Aha, ok, that's fine by me. But I'm a little confused by how all the nested copying works here. Does this actually cause the FGG to share its labeling with its rules? How does that work? (Sorry if I'm missing something obvious.)

Might be worth adding a comment here to explain that it does this.